src/directives/formly-field.js

```diff
@@ -87,9 +87,9 @@ function formlyField($http, $q, $compile, $templateCache, $interpolate, formlyCo
       if (!key || !model) {
         return
       }
+
       // If we are working with a number then $parse wont work, default back to the old way for now
-      if (angular.isNumber(key)) {
-        // TODO: Fix this so we can get several levels instead of just one with properties that are numeric
+      if (angular.isNumber(key) || (/\d/.test(key) && key.indexOf('[') === -1)) {
```

Comment:
Hmm…I stand corrected.  Your Regex foo is greater then mine. :-)

Ok…I’ll have to make the change.  Give me a bit though, I have burned through my playtime for the moment. :-)

From:  "Kent C. Dodds" <notifications@github.com>
Reply-To:  formly-js/angular-formly <reply@reply.github.com>
Date:  Thursday, December 3, 2015 at 1:00 PM
To:  formly-js/angular-formly <angular-formly@noreply.github.com>
Cc:  Matt Kruczek <kruz79@gmail.com>
Subject:  Re: [angular-formly] fix(formly-field): issue number 566 - Breaking change in 7.3.2 (#569)


In src/directives/formly-field.js:
>        // If we are working with a number then $parse wont work, default back to the old way for now
> -      if (angular.isNumber(key)) {
> -        // TODO: Fix this so we can get several levels instead of just one with properties that are numeric
> +      if (angular.isNumber(key) || (/\d/.test(key) && key.indexOf('[') === -1)) {
Yeah, that's what I'm saying. That's what /^\d/ would do. It simply would check whether the first character is a digit.

For the if statement, I would expect something like:

if (shouldNotUseParse(key) {
  return model[key]
} // etc.// function down below...function shouldNotUseParse(key) {
  return angular.isNumber(key) || (/^\d/.test(key) && key.indexOf('[') === -1)
}
—
Reply to this email directly or view it on GitHub.